internal/client/ierror.go

```diff
@@ -0,0 +1,15 @@
+package client
+
+import "fmt"
+
+type iError struct {
+	status string
+}
+
+func newIError(status string) *iError {
```

Comment:
method to create new instance should be exported `NewIError`

Reply:
but i design it for internal use only :D